packages/grafana-data/src/types/logs.ts

```diff
@@ -168,11 +170,32 @@ export interface DataSourceWithLogsContextSupport<TQuery extends DataQuery = Dat
 }
 
 export const hasLogsContextSupport = (datasource: unknown): datasource is DataSourceWithLogsContextSupport => {
-  if (!datasource) {
-    return false;
+  if (datasource && typeof datasource === 'object') {
+    const hasGetLogRowContext = 'getLogRowContext' in datasource && datasource.getLogRowContext !== undefined;
+    const hasShowContextToggle = 'showContextToggle' in datasource && datasource.showContextToggle !== undefined;
+
+    return hasGetLogRowContext && hasShowContextToggle;
   }
 
-  const withLogsSupport = datasource as DataSourceWithLogsContextSupport;
+  return false;
+};
 
-  return withLogsSupport.getLogRowContext !== undefined && withLogsSupport.showContextToggle !== undefined;
+/**
+ * Data sources that support log volume.
+ * This will enable full-range log volume histogram in Explore.
+ */
+export interface DataSourceWithLogsVolumeSupport<TQuery extends DataQuery> {
+  getLogsVolumeDataProvider(request: DataQueryRequest<TQuery>): Observable<DataQueryResponse> | undefined;
+}
+
+export const hasLogsVolumeSupport = <TQuery extends DataQuery>(
```

Comment:
I'm getting this when running `yarn start` 🤔

```
ERROR in ./.yarn/__virtual__/@grafana-data-virtual-adf1e43f85/1/packages/grafana-data/src/types/logs.ts:196:63
TS2339: Property 'getLogsVolumeDataProvider' does not exist on type 'object'.
    194 |   if (datasource && typeof datasource === 'object') {
    195 |     const hasGetLogsVolumeDataProvider =
  > 196 |       'getLogsVolumeDataProvider' in datasource && datasource.getLogsVolumeDataProvider !== undefined;
        |                                                               ^^^^^^^^^^^^^^^^^^^^^^^^^
    197 |
    198 |     return hasGetLogsVolumeDataProvider;
    199 |   }
```

Reply:
Yes, this is because it depends on new typescript version 4.9.3, which was reverted. I am going to change this to draft, and reopen when the version is updated to 4.9.3.